### PR TITLE
Guvnor 3125

### PR DESCRIPTION
**Pull request by desmax74**
Classes moved from old guvnor repo plus MaxB updated by APorcelli

### CODE REVIEW

File: uberfire-project/uberfire-project-api/src/main/java/org/guvnor/common/services/project/builder/model/BuildMessage.java [hunk 1]
```diff
@@ -17,13 +17,14 @@
...
 import java.io.Serializable;
+import java.util.Comparator;
 
 import org.guvnor.common.services.shared.message.Level;
 import org.jboss.errai.common.client.api.annotations.Portable;
 import org.uberfire.backend.vfs.Path;
 
 @Portable
-public class BuildMessage implements Serializable {
+public class BuildMessage implements Serializable, Comparable {
```
Comment (jhrcek) [inline uberfire-project/uberfire-project-api/src/main/java/org/guvnor/common/services/project/builder/model/BuildMessage.java]: Please avoid using raw types. Using `Comparable<BuildMessage>` is more type safe and you can avoid casting in compareTo implementation:

```
    @Override
    public int compareTo(BuildMessage other) {
        return Comparator.comparing(BuildMessage::getText).compare(this, other);
    }
```

The same comment applies to `ValidationMessage` below

File: uberfire-project/uberfire-project-backend/src/test/java/org/guvnor/common/services/project/backend/server/utils/ConfigurationStaticStrategy.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                                                    Order {
+
+    protected Map<ConfigurationKey, String> conf;
+
+    private Boolean valid = Boolean.FALSE;
+
+    public ConfigurationStaticStrategy() {
+        conf = new HashMap<>();
+
+        conf.put(ConfigurationKey.MAVEN_PLUGINS,
```
Comment (jhrcek) [inline uberfire-project/uberfire-project-backend/src/test/java/org/guvnor/common/services/project/backend/server/utils/ConfigurationStaticStrategy.java]: Could we maybe rename these enum constants to make it clear what each of them means?

MAVEN_COMPILER_PLUGIN_GROUP
MAVEN_COMPILER_PLUGIN_ARTIFACT
MAVEN_COMPILER_PLUGIN_VERSION

and similar for Takari alternative?

**Review by jhrcek — CHANGES_REQUESTED** [review]

**Review by jhrcek — APPROVED** [review]
Thanks for the changes!

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java [hunk 3]
```diff
@@ -47,33 +86,126 @@ public String toString(final POM pomModel)
...
+        String kieVersion = conf.get(ConfigurationKey.KIE_VERSION);
+        if (pom.getDependencies().isEmpty()) {
+            List<Dependency> dependencies = new ArrayList<>();
+            dependencies.add(getDependency(conf, "org.kie", "kie-api", kieVersion, "provided"));
+            dependencies.add(getDependency(conf, "org.optaplanner", "optaplanner-core", kieVersion, "provided"));
+            dependencies.add(getDependency(conf, "org.optaplanner", "org.optaplanner", kieVersion, "provided"));
+            dependencies.add(getDependency(conf, "junit", "junit", "4.12", "test"));
+            model.setDependencies(dependencies);
+        } else {
+            pom.getDependencies().add(getGuvDependency(conf, "org.kie", "kie-api", kieVersion, "provided"));
```
Comment (jhrcek) [inline uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java]: Is there a reason to add `org.kie:kie-api` dependency twice or did you want to add some other dependency?

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java [hunk 4]
```diff
@@ -47,33 +86,126 @@ public String toString(final POM pomModel)
...
+            pom.getDependencies().add(getGuvDependency(conf, "org.kie", "kie-api", kieVersion, "provided"));
+            pom.getDependencies().add(getGuvDependency(conf, "org.optaplanner", "optaplanner-core", kieVersion, "provided"));
+            pom.getDependencies().add(getGuvDependency(conf, "org.optaplanner", "org.optaplanner", kieVersion, "provided"));
+            pom.getDependencies().add(getGuvDependency(conf, "junit", "junit", "4.12", "test"));
+            new DependencyUpdater(model.getDependencies()).updateDependencies(pom.getDependencies());
+        }
+    }
+
+    private Dependency getDependency(Map<ConfigurationKey, String> conf, String groupID, String artifactID, String version, String scope) {
```
Comment (jhrcek) [inline uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java]: The conf Map parameter is unused in this private method. Please remove it.

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java [hunk 5]
```diff
@@ -47,33 +86,126 @@ public String toString(final POM pomModel)
...
+            dependencies.add(getDependency(conf, "org.kie", "kie-api", kieVersion, "provided"));
+            dependencies.add(getDependency(conf, "org.optaplanner", "optaplanner-core", kieVersion, "provided"));
+            dependencies.add(getDependency(conf, "org.optaplanner", "org.optaplanner", kieVersion, "provided"));
+            dependencies.add(getDependency(conf, "junit", "junit", "4.12", "test"));
+            model.setDependencies(dependencies);
+        } else {
+            pom.getDependencies().add(getGuvDependency(conf, "org.kie", "kie-api", kieVersion, "provided"));
+            pom.getDependencies().add(getGuvDependency(conf, "org.kie", "kie-api", kieVersion, "provided"));
+            pom.getDependencies().add(getGuvDependency(conf, "org.optaplanner", "optaplanner-core", kieVersion, "provided"));
+            pom.getDependencies().add(getGuvDependency(conf, "org.optaplanner", "org.optaplanner", kieVersion, "provided"));
```
Comment (jhrcek) [inline uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java]: This second optaplanner dependency looks like a mistake too. Is there any dep with
GroupId = org.optaplanner and 
ArtifactId = ord.optaplanner?

**Review by jhrcek — COMMENTED** [review]
I'm writing some tests to expand coverage of POMContentHandler, but there are some problems in the dependencies being added. If you could fix / answer those I'll extend the coverage..those

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java (hunk 4 above)
Comment (desmax74) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java]: fixed

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java (hunk 5 above)
Comment (desmax74) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java]: fixed

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java (hunk 3 above)
Comment (desmax74) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/POMContentHandler.java]: fixed

**Comment by ederign** [comment]
Can you please move this PR for https://github.com/kiegroup/appformer ?

**Comment by desmax74** [comment]
No more
relevant after two years

**Pull request closed without merging**